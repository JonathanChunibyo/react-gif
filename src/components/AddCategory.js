import React, { useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({ setcategory }) => {
    const vacio = () => {
        setinputValue('');
    }
    const [inputValue, setinputValue] = useState('');
    const handleInputvalue = (e) => {
        setinputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2) {
            setcategory(cats => [inputValue, ...cats]);
            setinputValue('');
        }
    }
    return (

        <form onSubmit={handleSubmit}>

            <h1> {inputValue}</h1>
            <input
                type="text"
                value={inputValue}
                onChange={handleInputvalue}
                onClick={vacio}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setcategory: PropTypes.func.isRequired ,
}

