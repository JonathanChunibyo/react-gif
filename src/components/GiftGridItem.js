import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({title, url}) => {
    return (
        <div className='card animate__animated animate__fadeInLeft'>
            <p>{title}</p>
            <img src= {url} alt={title}/>
        </div>
    )
}
GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired ,
    url: PropTypes.string.isRequired,
}
