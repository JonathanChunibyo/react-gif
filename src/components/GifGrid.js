import React /* { useEffect, useState } */ from 'react'
import { useFetchGif } from '../hooks/useFetchGif'
//import { getGif } from '../helpers/getGifs';
import { GiftGridItem } from './GiftGridItem';
import PropTypes from "prop-types";


export const GifGrid = ({ category }) => {


    const {data: images, loading} = useFetchGif(category);


    
    
    /* const [images, setimages] = useState([]);
    useEffect(() => {
        getGif(category).then( setimages );
    }, [category]);
 */

    return (
        <>

            <h3 className='animate__animated animate__zoomInDown'> {category} </h3>
            {loading && <p className='animate__animated animate__shakeX'>'Cargando...'</p>}
            
            <div className='cardgrid'>
                {
                     //images.map(img => (
                       // <li key={img.id}>{img.title}</li>
                    //)) 
                    images.map((img) =>
                        <GiftGridItem
                            key={img.id}
                            {...img}
                        />)
                }
            </div>
        </>
    )
}
GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}