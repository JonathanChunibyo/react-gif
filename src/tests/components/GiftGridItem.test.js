import { GiftGridItem } from "../../components/GiftGridItem";
import React from 'react';
import { shallow } from 'enzyme';


describe('prueba GiftGridItem', () => {
    const title = 'Hola soy un titulo';
    const url = 'www.facebook.com'
    let wraper = shallow(<GiftGridItem title={title} url= {url} />);

    test(' verificar con snapshots los cambios que reciba el aplicativo ', () => {
        expect(wraper).toMatchSnapshot();
    })

    test('debe de tener un parrafo con el title', () => {
        const p = wraper.find('p');
        expect(p.text().trim()).toBe(title);
    })

    test('debe tener una imagen con url y alt igual a los props', () => {
        const img = wraper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })
    test('div debe tener la clase de animacion', () => {
        const div = wraper.find('div');

        
        const className = div.prop('className');
        expect(className.includes('animate__animated animate__fadeInLeft')).toBe(true);
    })
    
    


})
