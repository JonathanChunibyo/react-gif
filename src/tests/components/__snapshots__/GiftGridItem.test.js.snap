// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba GiftGridItem  verificar con snapshots los cambios que reciba el aplicativo  1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <p>
    Hola soy un titulo
  </p>
  <img
    alt="Hola soy un titulo"
    src="www.facebook.com"
  />
</div>
`;
